src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compFunc =
+      compareFunction ||
+      function (a, b) {
+        const aStr = String(a);
+        const bStr = String(b);
+
+        if (aStr < bStr) {
+          return -1;
+        }
+
+        if (aStr > bStr) {
+          return 1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 1; i < this.length; i++) {
+      const current = this[i];
+      let j = i - 1;
+
+      while (j >= 0 && compFunc(this[j], current) > 0) {
+        this[j + 1] = this[j];
+        j--;
+      }
+      this[j + 1] = current;
+    }
+
+    return this;
```

Comment:
It's good to see that you've implemented a custom sort method. However, the algorithm used here is an insertion sort, not a bubble sort. While insertion sort is fine for small arrays or arrays that are mostly sorted, it can be inefficient for larger, unsorted arrays. Consider implementing a more efficient sorting algorithm for better performance on various datasets.